### PR TITLE
blockdev: get path directly from lsblk output

### PR DESCRIPTION
**Pull request by nikita-dubrovskii**
Signed-off-by: Nikita Dubrovskii <nikita@linux.ibm.com>

### CODE REVIEW

File: src/blockdev.rs [hunk 1]
```diff
@@ -72,8 +72,10 @@ struct BlkDev {
 
 fn get_partitions(device: &str) -> Result<Vec<BlkDev>> {
     // have lsblk enumerate partitions on the device
+    // Older lsblk, e.g. in CentOS 7.6, doesn't support PATH, but -p option
     let result = Command::new("lsblk")
         .arg("--pairs")
+        .arg("-p")
```
Comment (jlebon) [inline src/blockdev.rs:78]: Can we use `--paths` here instead to be more clear?

File: src/blockdev.rs [hunk 2]
```diff
@@ -91,27 +93,13 @@ fn get_partitions(device: &str) -> Result<Vec<BlkDev>> {
...
-            .parent()
-            .chain_err(|| format!("couldn't get parent directory of {}", device))?
-            .to_path_buf();
-        match fields.get("NAME") {
-            None => continue,
-            Some(name) => path.push(name),
-        }
-        // Skip the device itself
-        if path == Path::new(device) {
-            continue;
```
Comment (jlebon) [inline src/blockdev.rs:107]: I think we still need some logic like this though, but I guess comparing to `/dev/${device}` instead?

**Comment by jlebon** [comment]
Could you add some information in the commit message about what issue this patch is resolving and why? Providing context is important both for reviewers and for posterity. And maybe a commit message title like: `blockdev: get path directly from lsblk output` or similar?

File: src/blockdev.rs (hunk 2 above)
Comment (nikita-dubrovskii) [reply src/blockdev.rs:107]: I guess `device` always contains `/dev/` prefix.
But i don't think we really need to filter out device itself, the only reason to do it is to fail with `couldn't find any partitions on {device}`, i think it's better to fail with `couldn't find {} device for {label}`

File: src/blockdev.rs (hunk 1 above)
Comment (nikita-dubrovskii) [reply src/blockdev.rs:78]: agree. done

**Comment by jlebon** [comment]
Gonna merge this one. This also fixes an issue where if you pass a `/dev/disk/by-X/...` disk path, it builds a partition path like `/dev/disk/by-X/${NAME}` which is wrong in this case. Using `--paths` prints the right path.

**Pull request merged**